crates/ruff_python_formatter/src/expression/expr_slice.rs

```diff
@@ -175,17 +177,16 @@ pub(crate) fn find_colons(
     let after_upper = upper
         .as_ref()
         .map_or(first_colon.end(), |upper| upper.range().end());
-    // At least the closing bracket must exist, so there must be a token there
-    let next_token =
-        first_non_trivia_token(after_upper, contents).ok_or(FormatError::SyntaxError {
-            message: "Din't find any token for slice second colon",
-        })?;
-    let second_colon = if next_token.kind == TokenKind::Colon {
-        debug_assert!(
-            next_token.range.start() < range.end(),
-            "The next token in a slice must either be a colon or the closing bracket"
-        );
-        Some(next_token)
+    let mut tokens = SimpleTokenizer::new(contents, TextRange::new(after_upper, range.end()))
+        .skip_trivia()
+        .skip_while(|token| token.kind == TokenKind::RParen);
```

Comment:
I wonder if we should create a helper for this. It's not the first time we incorrectly assumed that there are never right parens. But I don't have a good idea of how, and this is cleary out of this PR's scope.

Reply:
An after node range search util would be nice. I did some small changes in #5885